.github/workflows/extension_upgrade.yml

```diff
@@ -27,7 +27,7 @@ jobs:
     name: Upgrade Test
     runs-on: ubuntu-22.04
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: Install Rust stable toolchain
         uses: actions-rs/toolchain@v1
```

Comment:
CI also warns about `actions-rs/toolchain@v1`, but it was [`archived by the owner on Oct 13, 2023`](https://github.com/actions-rs/toolchain).

Fixing that may be involved enough to warrant its own pull request.